### ISSUE TITLE
Overriding app config for integration test project does not work based on the docu you've created

### ISSUE THREAD

**Issue by an2ny** [dotnet/AspNetCore.Docs#25364]
i have this startup

```
using My.API;
using My.Infrastructure;
using My.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.
builder.Services.AddApplication(builder.Configuration);
builder.Services.AddInfrastructure(builder.Configuration);//register db context here along with getting the connection string from config

builder.Services.AddControllers();
// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
builder.Services.AddEndpointsApiExplorer();
builder.Services.AddSwaggerGen();

var app = builder.Build();

// Configure the HTTP request pipeline.
if (app.Environment.IsDevelopment())
{
    app.UseSwagger();
    app.UseSwaggerUI();
}

app.InitializeDatabase(app.Services);
app.UseHttpsRedirection();
app.UseAuthorization();
app.MapControllers();

app.Run();

public partial class Program { }
```

Now i am trying to use a different appsettings.json for the testing project but doesn't seem to work correctly when implemented this way.

```
public class TestApplication : WebApplicationFactory<Program>
    {
        public IServiceScopeFactory ScopeFactory = null!;
        public IConfiguration Configuration = null!;

        protected override IHost CreateHost(IHostBuilder builder)
        {
            builder.ConfigureAppConfiguration((context, config) =>
            {
                config.SetBasePath(Directory.GetCurrentDirectory());
                config.AddJsonFile("appsettings.json");
                config.AddEnvironmentVariables();
            });

            builder.ConfigureServices(services =>
            {
                var descriptor = services.SingleOrDefault(d => d.ServiceType == typeof(DbContextOptions<ApplicationDbContext>));

                if (descriptor != null)
                    services.Remove(descriptor);

                var sp = services.BuildServiceProvider();
                var config = sp.GetRequiredService<IConfiguration>();
                var connectionString = config["ConString"];
                var serverVersion = ServerVersion.AutoDetect(connectionString);

                services.AddDbContext<ApplicationDbContext>(dbContextOptions =>
                    dbContextOptions
                        .UseMySql(connectionString, serverVersion)
                );
                services.AddScoped<IApplicationDbContext>(provider => provider.GetRequiredService<ApplicationDbContext>());
                ScopeFactory = sp.GetRequiredService<IServiceScopeFactory>();

                using (var scope = sp.CreateScope())
                {
                    var scopedServices = scope.ServiceProvider;
                    var db = scopedServices.GetRequiredService<ApplicationDbContext>();

                    db.Database.Migrate();
                }
            });

            return base.CreateHost(builder);
        }
```

i had to spend a couple of days in tears/agony/anger 😆  only to find out that it would work if it is like this.

```
public class TestApplication : WebApplicationFactory<Program>
    {
        public IServiceScopeFactory ScopeFactory = null!;
        public IConfiguration Configuration = null!;

        protected override IHost CreateHost(IHostBuilder builder)
        {
            builder.ConfigureServices(services =>
            {
                var descriptor = services.SingleOrDefault(d => d.ServiceType == typeof(DbContextOptions<ApplicationDbContext>));

                if (descriptor != null)
                    services.Remove(descriptor);

                var sp = services.BuildServiceProvider();
                var config = sp.GetRequiredService<IConfiguration>();
                var connectionString = config["ConString"];
                var serverVersion = ServerVersion.AutoDetect(connectionString);

                services.AddDbContext<ApplicationDbContext>(dbContextOptions =>
                    dbContextOptions
                        .UseMySql(connectionString, serverVersion)
                );
                services.AddScoped<IApplicationDbContext>(provider => provider.GetRequiredService<ApplicationDbContext>());
                ScopeFactory = sp.GetRequiredService<IServiceScopeFactory>();

                using (var scope = sp.CreateScope())
                {
                    var scopedServices = scope.ServiceProvider;
                    var db = scopedServices.GetRequiredService<ApplicationDbContext>();

                    db.Database.Migrate();
                }
            });

            builder.ConfigureHostConfiguration(config =>
            {
                config.SetBasePath(Directory.GetCurrentDirectory());
                config.AddJsonFile("appsettings.json");
                config.AddEnvironmentVariables();
            });

            return base.CreateHost(builder);
        }
```

had to use **`ConfigureHostConfiguration`** instead of **`ConfigureAppConfiguration`** and it worked and was able to override the appsettings.json.

i don't know but maybe it's just my project setup or whatever. but having correct information in your docu is a big thing. so i hope you get this correctly.
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 77889d43-fedb-f8b4-8fea-379ec946555c
* Version Independent ID: f499b131-8b14-7e10-1728-543fdd5a5656
* Content: [Integration tests in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-6.0#integration-tests-sample)
* Content Source: [aspnetcore/test/integration-tests.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/test/integration-tests.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-test**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
So the doc should be updated to say use **`ConfigureHostConfiguration`** instead of **`ConfigureAppConfiguration`**  ?

**Comment by Rick-Anderson** [comment]
This will be addressed in #27089
